Common/src/main/java/com/techarium/techarium/inventory/ExchangeStationMenu.java

```diff
@@ -0,0 +1,50 @@
+package com.techarium.techarium.inventory;
+
+import com.techarium.techarium.blockentity.selfdeploying.ExchangeStationBlockEntity;
+import com.techarium.techarium.registry.TechariumMenuTypes;
+import net.minecraft.core.BlockPos;
+import net.minecraft.world.SimpleContainer;
+import net.minecraft.world.entity.player.Inventory;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.inventory.Slot;
+import net.minecraft.world.item.ItemStack;
+
+public class ExchangeStationMenu extends MachineMenu {
+
+	private final SimpleContainer input;
+	private final SimpleContainer output;
+	private final ExchangeStationBlockEntity exchangeStation;
+
+	public ExchangeStationMenu(int id, Inventory playerInventory, Player player, BlockPos pos) {
+		super(TechariumMenuTypes.EXCHANGE_STATION.get(), id, pos);
+		this.exchangeStation = ((ExchangeStationBlockEntity) player.level.getBlockEntity(this.pos));
+
+		this.input = this.exchangeStation.getItemInput();
+		this.output = this.exchangeStation.getItemOutput();
+
+		this.addSlot(new Slot(input, 0, 99, 35));
+
+		this.addSlot(new OutputSlot(output, 0, 153, 35));
+
+		for (int y = 0; y < 3; y++) {
```

Comment:
This should be abstracted out as this will be used in multiple menus over and over with 1 or 2 values changed.